Source/FieldSolver/FiniteDifferenceSolver/MacroscopicProperties/MacroscopicProperties.cpp

```diff
@@ -161,30 +119,105 @@ MacroscopicProperties::InitData ()
     amrex::Print() << "we are in init data of macro \n";
     auto & warpx = WarpX::GetInstance();
 
-    IntVect Ex_stag = warpx.getEfield_fp(0,0).ixType().toIntVect();
-    IntVect Ey_stag = warpx.getEfield_fp(0,1).ixType().toIntVect();
-    IntVect Ez_stag = warpx.getEfield_fp(0,2).ixType().toIntVect();
-    IntVect Bx_stag = warpx.getBfield_fp(0,0).ixType().toIntVect();
-    IntVect By_stag = warpx.getBfield_fp(0,1).ixType().toIntVect();
-    IntVect Bz_stag = warpx.getBfield_fp(0,2).ixType().toIntVect();
+    // Get BoxArray and DistributionMap of warpx instant.
+    int lev = 0;
+    BoxArray ba = warpx.boxArray(lev);
+    DistributionMapping dmap = warpx.DistributionMap(lev);
+    const amrex::IntVect ng_EBSolver = warpx.getngE();
+    // Define material property multifabs using ba and dmap from WarpX instance
+    // sigma is cell-centered MultiFab
+    m_sigma_mf = std::make_unique<MultiFab>(ba, dmap, 1, ng_EBSolver);
+    // epsilon is cell-centered MultiFab
+    m_eps_mf = std::make_unique<MultiFab>(ba, dmap, 1, ng_EBSolver);
+    // mu is cell-centered MultiFab
+    m_mu_mf = std::make_unique<MultiFab>(ba, dmap, 1, ng_EBSolver);
```

Comment:
Just checking: Are we sure that `ba` used here, set as `ba = warpx.boxArray(lev)` above, is always cell-centered (so that these macroscopic MultiFabs are also cell-centered, as you say in the comments)? Or does one need to convert the index type in some cases?

Reply:
yes. this ba is always cell-centered. when conversion is needed we typically use convert(ba,nodal_flag)